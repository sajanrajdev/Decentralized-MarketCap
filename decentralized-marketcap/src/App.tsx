import React, { useEffect, useState } from "react";
import './App.css';
import { ApolloClient, InMemoryCache} from '@apollo/client';
import { ChainId, Token, Fetcher, Trade, Route, TokenAmount, TradeType, WETH } from '@uniswap/sdk'
import Tokentable from './Tokentable';
import {ETHER_PRICE, ALL_TOKENS} from './queries'
import {currencyFormatter} from './utils';
import Container from '@material-ui/core/Container';
import TextField from '@material-ui/core/TextField';
import MenuItem from '@material-ui/core/MenuItem';
import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import ButtonAppBar from './AppBar'
import { useWallet } from 'use-wallet'

interface TradeToken {
  name: string
  symbol: string
  address: string
  decimals: number
}

function App() {
  const [etherPrice, setEtherPrice] = useState<number>(0);
  const [tokenslist, setTokensList] = useState<any | any[]>([]);
  const [selectedKeys, setSelectedKeys] = useState<string>();
  const [token1, settoken1] = useState<TradeToken>({name: "", symbol: "", address: "", decimals: 0});
  const [token2, settoken2] = useState<TradeToken>({name: "", symbol: "", address: "", decimals: 0});
  const [selectToken1, setSelectToken1] = useState('');
  const [selectToken2, setSelectToken2] = useState('');
  const [inputToken1, setInputToken1] = useState('');
  const [inputToken2, setInputToken2] = useState('');

  const wallet = useWallet();

  const client = new ApolloClient({
    uri: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2',
    cache: new InMemoryCache()
  });

  const getEtherPrice = () => {
    client
    .query({
      query: ETHER_PRICE
    })
    .then(result => setEtherPrice(result.data.bundle.ethPrice.valueOf()));
    return(null);
  }

  const getAllTokens = () => {
    client
    .query({
      query: ALL_TOKENS
    })
    .then(result => setTokensList(result.data.tokens));
    return(null);
  }

  const sortTokenList = (tokenslist: any[], ethPrice: number) => {
    var sortedItems = tokenslist.map(
      token => ({...token, totalLiquidity: token.totalLiquidity.valueOf()*token.derivedETH.valueOf()*ethPrice, price: token.derivedETH.valueOf()*ethPrice})
    ); // create a new array of items with totalLiquidity and Price added
    sortedItems = sortedItems.sort((a,b) => a['totalLiquidity'] < b['totalLiquidity'] ? 1 : -1); //Sorts desc based on TotalLiquidity
    sortedItems = sortedItems.map(
      token => ({...token, totalLiquidity: currencyFormatter(token.totalLiquidity, 'usd'), price: currencyFormatter(token.price, 'usd')})
    ); // Format total liquidity and price to USD
    return sortedItems;
  }

  const getTokenBySymbol = (tokenslist: any[], selectedSymbol: string) => {
    var selectedToken = tokenslist.find(x => x.symbol === selectedSymbol)
        console.log(selectedToken);
      return (selectedToken);
    }

  const getTokensByID = (tokenslist: any[], selectedKeys: React.Key[] | React.Key) => {
    var selectedTokens: any[] = []
    let i: number = 0
    if(selectedKeys){
      selectedKeys.forEach(element => {
        selectedTokens[i] = tokenslist.find(x => x.id === element)
        i++;
      });
      return (selectedTokens);
    }
  }

  const HandleCheckBox = () => {
    if(selectedKeys){
      var tokens: any[] | any = getTokensByID(tokenslist, selectedKeys);
      if(selectedKeys.length == 1){
        setSelectToken1(tokens[0].symbol)
      }
      else if(selectedKeys.length == 2){
        setSelectToken2(tokens[1].symbol)
      }
      else if(selectedKeys.length >= 2){
        console.log("Toom many selected")
      }
    }
    return(null) 
  }

  // Get realtime price of token1 based on paired token2
  const getPrice = async (id1: string, decimals1: number, id2: string, decimals2: number) => {
    const token1 = new Token(ChainId.MAINNET, id1, decimals1);
    const token2 = new Token(ChainId.MAINNET, id2, decimals2);
    const pair = await Fetcher.fetchPairData(token1, token2);
    const route = new Route([pair], token1);
    const trade = new Trade(route, new TokenAmount(token1, '10000000000000000'), TradeType.EXACT_INPUT);
    console.log("Execution Price:", trade.executionPrice.toSignificant(6));
    console.log("Mid Price:", route.midPrice.toSignificant(6))
    return trade.executionPrice.toSignificant(6);
  }

  const BalanceButton = () => {
    return(
      <div>
      {wallet.status === 'connected' ? (
        <div>
          <Button variant="outlined" size="large" color="primary">
            <div>Balance: {(parseFloat(wallet.balance)/1000000000000000000)} ETH</div>
          </Button>
        </div>
      ) : (
        <div>
          <Button variant="outlined" size="large" color="primary" disabled>
            <div>Balance: Disconnected</div>
          </Button>
        </div>
      )}
      <br/>
      </div>
    )
  }

  // On Mount 
  useEffect(()=>{
    getEtherPrice();
    getAllTokens();
    sortTokenList(tokenslist, etherPrice);
  }, []);

  const handleChange1 = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSelectToken1(event.target.value);
    var token_temp = getTokenBySymbol(tokenslist, event.target.value);
    settoken1({name: token_temp.name, symbol: token_temp.symbol, address: token_temp.id, decimals: token_temp.decimals});
    setInputToken1('');
    setInputToken2('');
  };
  const handleInputChange1 = (event: React.ChangeEvent<HTMLInputElement>) => {
    console.log(event.target.value)
    setInputToken1(event.target.value);
    if(event.target.value==''){
      setInputToken2('');
    }
  };
  const handleChange2 = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSelectToken2(event.target.value);
    var token_temp = getTokenBySymbol(tokenslist, event.target.value);
    settoken2({name: token_temp.name, symbol: token_temp.symbol, address: token_temp.id, decimals: token_temp.decimals});
    setInputToken1('');
    setInputToken2('');
  };
  const handleEstimatePriceButton = async () => {
    var ExecutionPrice = await getPrice(token1.address, token1.decimals, token2.address, token2.decimals);
    setInputToken2((parseFloat(inputToken1)*parseFloat(ExecutionPrice)).toString());
  };

  return (
    <div className="App">
      <ButtonAppBar></ButtonAppBar>
      <header>
        <h1>
          Uniswap Remote Trader
        </h1>
      </header>

      <BalanceButton></BalanceButton>

      <Container>
        <HandleCheckBox></HandleCheckBox>
        <form className="form">
          <div>
            <TextField id="Select1" select label="Select" value={selectToken1} onChange={handleChange1} helperText="Please select your token 1" variant="outlined">
              {tokenslist.map((option: any | any[]) => (
                <MenuItem key={option.id} value={option.symbol}>
                  {option.symbol}
                </MenuItem>
              ))}
            </TextField>
            <TextField id="Input1" label="Amount" variant="outlined" value={inputToken1} color="primary" onChange={handleInputChange1} disabled={(selectToken1=='')||(selectToken2=='')} type="number"/>
          </div>
          <br/>
          <div>
            <TextField id="Select2" select label="Select" value={selectToken2} onChange={handleChange2} helperText="Please select your token 2" variant="outlined">
              {tokenslist.map((option: any | any[]) => (
                <MenuItem key={option.id} value={option.symbol}>
                  {option.symbol}
                </MenuItem>
              ))}
            </TextField>
            <TextField id="Input2" label="Estimated Execution Price" variant="outlined" value={inputToken2} color="primary" disabled type="number"/>
          </div>
        </form>
        <br/>
        <div>
        <ButtonGroup disableElevation variant="contained" color="primary">
          <Button variant="contained" size="large" color="primary" disabled={(inputToken1=='')} onClick={handleEstimatePriceButton}>
            Estimate
          </Button>
          <Button variant="contained" size="large" color="primary" disabled={(selectToken1!="WETH")||((parseFloat(wallet.balance)/1000000000000000000) < parseFloat(inputToken1))||(inputToken2=='')}>
            Swap
          </Button>
        </ButtonGroup>
        </div>
        <br/>

        <Tokentable coindata={sortTokenList(tokenslist, etherPrice)} selectRows={(selectedRowsKeys: string) => setSelectedKeys(selectedRowsKeys)}/>
      </Container>
    </div>
  );
}

export default App;
