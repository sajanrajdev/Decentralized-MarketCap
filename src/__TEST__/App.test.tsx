import React from 'react'
import ReactDOM from 'react-dom'
import App from '../App'

import Enzyme, { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {render, fireEvent, cleanup, getByTestId} from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() })

afterEach(cleanup);

// Snapshot test
/* it('renders correctly enzyme', () => {
    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot();
  }); */

it('Selecting tokens enables Input 1', () => {
    const { getByTestId, getByText } = render(<App />);
    fireEvent.change(getByTestId("Select1"), {target: {value: "WETH"}})
    fireEvent.change(getByTestId("Select2"), {target: {value: "UNI"}})
    expect(getByText(/Estimate/i).closest('Button')).not.toBeDisabled();
})

/* test("renders without crashing", () => {
    const root = document.createElement("div");
    ReactDOM.render(<App></App>, root);
}); */

