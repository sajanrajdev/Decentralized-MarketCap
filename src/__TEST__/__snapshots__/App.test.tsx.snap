// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "between": [Function],
        "down": [Function],
        "keys": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "only": [Function],
        "up": [Function],
        "values": Object {
          "lg": 1280,
          "md": 960,
          "sm": 600,
          "xl": 1920,
          "xs": 0,
        },
        "width": [Function],
      },
      "direction": "ltr",
      "mixins": Object {
        "gutters": [Function],
        "toolbar": Object {
          "@media (min-width:0px) and (orientation: landscape)": Object {
            "minHeight": 48,
          },
          "@media (min-width:600px)": Object {
            "minHeight": 64,
          },
          "minHeight": 56,
        },
      },
      "overrides": Object {},
      "palette": Object {
        "action": Object {
          "activatedOpacity": 0.24,
          "active": "#fff",
          "disabled": "rgba(255, 255, 255, 0.3)",
          "disabledBackground": "rgba(255, 255, 255, 0.12)",
          "disabledOpacity": 0.38,
          "focus": "rgba(255, 255, 255, 0.12)",
          "focusOpacity": 0.12,
          "hover": "rgba(255, 255, 255, 0.08)",
          "hoverOpacity": 0.08,
          "selected": "rgba(255, 255, 255, 0.16)",
          "selectedOpacity": 0.16,
        },
        "augmentColor": [Function],
        "background": Object {
          "default": "#303030",
          "paper": "#424242",
        },
        "common": Object {
          "black": "#000",
          "white": "#fff",
        },
        "contrastThreshold": 3,
        "divider": "rgba(255, 255, 255, 0.12)",
        "error": Object {
          "contrastText": "#fff",
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "getContrastText": [Function],
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
          "A100": "#d5d5d5",
          "A200": "#aaaaaa",
          "A400": "#303030",
          "A700": "#616161",
        },
        "info": Object {
          "contrastText": "#fff",
          "dark": "#1976d2",
          "light": "#64b5f6",
          "main": "#2196f3",
        },
        "primary": Object {
          "contrastText": "#fff",
          "dark": "rgb(170, 14, 87)",
          "light": "rgb(246, 67, 151)",
          "main": "#F4157D",
        },
        "secondary": Object {
          "contrastText": "#fff",
          "dark": "rgb(100, 7, 51)",
          "light": "rgb(165, 59, 109)",
          "main": "#8f0b49",
        },
        "success": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "#388e3c",
          "light": "#81c784",
          "main": "#4caf50",
        },
        "text": Object {
          "disabled": "rgba(255, 255, 255, 0.5)",
          "hint": "rgba(255, 255, 255, 0.5)",
          "icon": "rgba(255, 255, 255, 0.5)",
          "primary": "#fff",
          "secondary": "rgba(255, 255, 255, 0.7)",
        },
        "tonalOffset": 0.2,
        "type": "dark",
        "warning": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "#f57c00",
          "light": "#ffb74d",
          "main": "#ff9800",
        },
      },
      "props": Object {},
      "shadows": Array [
        "none",
        "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
        "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
        "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
        "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
        "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
        "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
        "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
        "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
        "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
        "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
        "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
        "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
        "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
        "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
        "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
        "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
        "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
        "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
        "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
        "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
        "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
        "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
        "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
        "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
      ],
      "shape": Object {
        "borderRadius": 4,
      },
      "spacing": [Function],
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
        "getAutoHeightDuration": [Function],
      },
      "typography": Object {
        "body1": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": 1.5,
        },
        "body2": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "0.875rem",
          "fontWeight": 400,
          "lineHeight": 1.43,
        },
        "button": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": 1.75,
          "textTransform": "uppercase",
        },
        "caption": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 1.66,
        },
        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
        "fontSize": 14,
        "fontWeightBold": 700,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "h1": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "6rem",
          "fontWeight": 300,
          "lineHeight": 1.167,
        },
        "h2": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "3.75rem",
          "fontWeight": 300,
          "lineHeight": 1.2,
        },
        "h3": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "3rem",
          "fontWeight": 400,
          "lineHeight": 1.167,
        },
        "h4": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "2.125rem",
          "fontWeight": 400,
          "lineHeight": 1.235,
        },
        "h5": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "lineHeight": 1.334,
        },
        "h6": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "1.25rem",
          "fontWeight": 500,
          "lineHeight": 1.6,
        },
        "htmlFontSize": 16,
        "overline": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "lineHeight": 2.66,
          "textTransform": "uppercase",
        },
        "pxToRem": [Function],
        "round": [Function],
        "subtitle1": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "1rem",
          "fontWeight": 400,
          "lineHeight": 1.75,
        },
        "subtitle2": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", \\"sans-serifSegoe UI\\", \\"Oxygen\\",
      \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\"",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "lineHeight": 1.57,
        },
      },
      "zIndex": Object {
        "appBar": 1100,
        "drawer": 1200,
        "mobileStepper": 1000,
        "modal": 1300,
        "snackbar": 1400,
        "speedDial": 1050,
        "tooltip": 1500,
      },
    }
  }
>
  <div
    className="App"
    data-testid="App"
  >
    <WithStyles(ForwardRef(Paper))>
      <TopAppBar
        address={null}
        darkmode={true}
        onChange={[Function]}
      />
      <br />
      <header>
        <WithStyles(ForwardRef(Typography))
          align="center"
          variant="h3"
        >
          Uniswap Remote Trader
        </WithStyles(ForwardRef(Typography))>
      </header>
      <br />
      <WithStyles(ForwardRef(Container))>
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          direction="column"
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Paper))
              className="makeStyles-paper-1"
              elevation={3}
              style={
                Object {
                  "height": 400,
                  "width": 550,
                }
              }
            >
              <Styled(MuiBox)
                m={1}
                p={1}
              >
                <WithStyles(ForwardRef(Grid))
                  alignItems="center"
                  container={true}
                  direction="column"
                  justify="center"
                  spacing={2}
                >
                  <WithStyles(ForwardRef(Grid))
                    alignItems="center"
                    container={true}
                    direction="row"
                    item={true}
                    justify="flex-end"
                    spacing={2}
                  >
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      Balance:
                    </WithStyles(ForwardRef(Grid))>
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      <BalanceButton
                        selectToken1=""
                        setInputToken1={[Function]}
                        setInputToken2={[Function]}
                        wallet={Object {}}
                      />
                    </WithStyles(ForwardRef(Grid))>
                  </WithStyles(ForwardRef(Grid))>
                  <WithStyles(ForwardRef(Grid))
                    container={true}
                    direction="row"
                    item={true}
                    justify="center"
                    spacing={2}
                  >
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      <WithStyles(ForwardRef(TextField))
                        helperText="From"
                        id="Select1"
                        label="Token"
                        onChange={[Function]}
                        select={true}
                        style={
                          Object {
                            "width": 230,
                          }
                        }
                        value=""
                        variant="outlined"
                      >
                        <WithStyles(ForwardRef(MenuItem))
                          key="0xc778417e063141139fce010982780140aa0cd5ab"
                          value="WETH"
                        >
                          WETH
                        </WithStyles(ForwardRef(MenuItem))>
                      </WithStyles(ForwardRef(TextField))>
                    </WithStyles(ForwardRef(Grid))>
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      <WithStyles(ForwardRef(TextField))
                        color="primary"
                        disabled={true}
                        error={false}
                        id="Input1"
                        label="Amount"
                        onChange={[Function]}
                        placeholder="0.0"
                        style={
                          Object {
                            "width": 230,
                          }
                        }
                        type="number"
                        value=""
                        variant="outlined"
                      />
                    </WithStyles(ForwardRef(Grid))>
                  </WithStyles(ForwardRef(Grid))>
                  <WithStyles(ForwardRef(Grid))
                    container={true}
                    direction="row"
                    item={true}
                    justify="center"
                    spacing={2}
                  >
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      <WithStyles(ForwardRef(TextField))
                        helperText="To"
                        id="Select2"
                        label="Token"
                        onChange={[Function]}
                        select={true}
                        style={
                          Object {
                            "width": 230,
                          }
                        }
                        value=""
                        variant="outlined"
                      >
                        <WithStyles(ForwardRef(MenuItem))
                          key="0x1f9840a85d5af5bf1d1762f925bdaddc4201f984"
                          value="UNI"
                        >
                          UNI
                        </WithStyles(ForwardRef(MenuItem))>
                        <WithStyles(ForwardRef(MenuItem))
                          key="0xc7ad46e0b8a400bb3c915120d284aafba8fc4735"
                          value="DAI"
                        >
                          DAI
                        </WithStyles(ForwardRef(MenuItem))>
                      </WithStyles(ForwardRef(TextField))>
                    </WithStyles(ForwardRef(Grid))>
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      <WithStyles(ForwardRef(TextField))
                        color="primary"
                        disabled={true}
                        id="Input2"
                        label="Estimated Price"
                        placeholder="0.0"
                        style={
                          Object {
                            "width": 230,
                          }
                        }
                        type="number"
                        value=""
                        variant="outlined"
                      />
                    </WithStyles(ForwardRef(Grid))>
                  </WithStyles(ForwardRef(Grid))>
                </WithStyles(ForwardRef(Grid))>
                <WithStyles(ForwardRef(Grid))
                  alignItems="center"
                  container={true}
                  direction="column"
                  spacing={2}
                >
                  <WithStyles(ForwardRef(Grid))
                    item={true}
                  >
                    <WithStyles(ForwardRef(Typography))
                      gutterBottom={true}
                      id="discrete-slider-small-steps"
                    >
                      Slippage Tolerance
                    </WithStyles(ForwardRef(Typography))>
                    <WithStyles(ForwardRef(Slider))
                      aria-labelledby="discrete-slider-small-steps"
                      defaultValue={0.5}
                      marks={true}
                      max={1}
                      min={0}
                      onChangeCommitted={[Function]}
                      step={0.1}
                      valueLabelDisplay="auto"
                    />
                  </WithStyles(ForwardRef(Grid))>
                  <WithStyles(ForwardRef(Grid))
                    container={true}
                    direction="row"
                    item={true}
                    justify="center"
                    spacing={5}
                  >
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      <WithStyles(ForwardRef(TextField))
                        color="primary"
                        error={false}
                        helperText="Gas Price"
                        id="Gas Price"
                        label=""
                        onChange={[Function]}
                        placeholder="20"
                        size="small"
                        style={
                          Object {
                            "width": 60,
                          }
                        }
                        type="number"
                        value="20"
                        variant="standard"
                      />
                      GWei
                    </WithStyles(ForwardRef(Grid))>
                    <WithStyles(ForwardRef(Grid))
                      item={true}
                    >
                      <WithStyles(ForwardRef(TextField))
                        color="primary"
                        error={false}
                        helperText="Deadline"
                        id="Deadline"
                        label=""
                        onChange={[Function]}
                        placeholder="20"
                        size="small"
                        style={
                          Object {
                            "width": 50,
                          }
                        }
                        type="number"
                        value="20"
                        variant="standard"
                      />
                      min
                    </WithStyles(ForwardRef(Grid))>
                  </WithStyles(ForwardRef(Grid))>
                </WithStyles(ForwardRef(Grid))>
              </Styled(MuiBox)>
            </WithStyles(ForwardRef(Paper))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          />
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(ButtonGroup))
              color="primary"
              disableElevation={true}
              variant="contained"
            >
              <WithStyles(ForwardRef(Button))
                color="primary"
                disabled={true}
                onClick={[Function]}
                size="large"
                variant="contained"
              >
                Estimate
              </WithStyles(ForwardRef(Button))>
              <WithStyles(ForwardRef(Button))
                color="primary"
                disabled={true}
                onClick={[Function]}
                size="large"
                variant="contained"
              >
                Swap
              </WithStyles(ForwardRef(Button))>
            </WithStyles(ForwardRef(ButtonGroup))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <br />
        <Tokentable
          coindata={Array []}
        />
      </WithStyles(ForwardRef(Container))>
      <br />
      <div>
        <WithStyles(ForwardRef(Typography))
          align="center"
          variant="body2"
        >
          © 2021 Sajan Rajdev. All Rights Reserved.
        </WithStyles(ForwardRef(Typography))>
      </div>
      <br />
    </WithStyles(ForwardRef(Paper))>
  </div>
</ThemeProvider>
`;
